elastic/kibana
# [Fleet] Handler api key creation errors when Fleet Admin is invalid

## Summary

Resolve https://github.com/elastic/kibana/issues/84412

We use an user `fleet_enroll` stored in a saved object to create API keys in Fleet. Currently if this user is updated (like a password change) outside of Fleet, our APIs that used that user will return a 401 and this will cause the current User to be logged out from Kibana.

This PR fix this by handling that error and returning a 400 with the message `Fleet Admin user is invalid`  also we clear the cache we use to store the `fleet_enroll` user.

## How to test this?

1. Setup Fleet and Agents
1. Modify the fleet_enroll user  `/app/management/security/users/edit/fleet_enroll`
<img width="805" alt="Screen Shot 2020-11-30 at 2 43 01 PM" src="https://user-images.githubusercontent.com/1336873/100656933-7bdac300-331b-11eb-8b89-8c723f0ddd8d.png">

1. Try to create a new enrollment API key
 - Before this caused the user to logged out
 - Now we should have this error message from the API
<img width="906" alt="Screen Shot 2020-11-30 at 2 32 20 PM" src="https://user-images.githubusercontent.com/1336873/100656908-72e9f180-331b-11eb-88a4-c641e9b5c7f5.png">
